llvm/lib/IR/Metadata.cpp

```diff
@@ -1620,10 +1629,7 @@ void Instruction::setNoSanitizeMetadata() {
 }
 
 MDNode *Instruction::getMetadataImpl(unsigned KindID) const {
-  // Handle 'dbg' as a special case since it is not stored in the hash table.
-  if (KindID == LLVMContext::MD_dbg)
-    return DbgLoc.getAsMDNode();
-  return Value::getMetadata(KindID);
+  return getValueMetadata(*this, KindID, getContext());
```

Comment:
Could you leave a comment here indicting the assumptions of this method (non-'dbg', already checked metadata is present), it'll make it easier for those who aren't familiar with the Metadata hierarchy to understand what's going on.

Reply:
On that note, does this function even need to exist still? Or should callers directly call getValueMetadata? If it's useful, should it be renamed getValueMetadata (an overload with fewer parameters)?

Reply:
Shuffled the functions around a bit and changed `getValueMetada` to a new `Value::getMetadataImpl` and removed `Instruction::getMetadataImpl` now.